README.md
# Colours

An extended version of Flutter Colors with more swatches and more flexibility to generate your own custom swatch.

## Getting Started

In your flutter project add the dependency:

```
dependencies:
  colour: ^0.0.2
```

or reference the git repo directly:

```
dependencies:
  colour:
    git: 
      url: git://github.com/salkuadrat/colours.git
```

## Examples

Suppose your brand color is `#00aa13`.

![#00aa13](https://via.placeholder.com/15/00aa13/000000?text=+) `Gojek Brand Color [#00aa13]`

You can generate your own custom swatch using:

```
MaterialColor gojekSwatch = Colours.swatch('#00aa13');
```

It will generate a complete material swatch colors:

- ![#e0f5e3](https://via.placeholder.com/15/e0f5e3/000000?text=+) `50  [#e0f5e3]`
- ![#b2e5b8](https://via.placeholder.com/15/b2e5b8/000000?text=+) `100 [#b2e5b8]`
- ![#7fd489](https://via.placeholder.com/15/7fd489/000000?text=+) `200 [#7fd489]`
- ![#4cc45a](https://via.placeholder.com/15/4cc45a/000000?text=+) `300 [#4cc45a]`
- ![#26b736](https://via.placeholder.com/15/26b736/000000?text=+) `400 [#26b736]`
- ![#00aa13](https://via.placeholder.com/15/00aa13/000000?text=+) `500 [#00aa13]`
- ![#00a311](https://via.placeholder.com/15/00a311/000000?text=+) `600 [#00a311]`
- ![#00990e](https://via.placeholder.com/15/00990e/000000?text=+) `700 [#00990e]`
- ![#00900b](https://via.placeholder.com/15/00900b/000000?text=+) `800 [#00900b]`
- ![#007f06](https://via.placeholder.com/15/007f06/000000?text=+) `900 [#007f06]`


You can apply the custom swatch to Theme, or access the individual color directly using:

```
Color gojek100 = gojekSwatch[100];
Color gojek200 = gojekSwatch[200];
Color gojek300 = gojekSwatch[300];
Color gojek400 = gojekSwatch[400];
Color gojek500 = gojekSwatch[500];
Color gojek600 = gojekSwatch[600];
Color gojek700 = gojekSwatch[700];
Color gojek800 = gojekSwatch[800];
Color gojek900 = gojekSwatch[900];
```

## Custom Pre-Generated Swatches

Class Colours also come with custom pre-generated swatches. 

- ![#f0f8ff](https://via.placeholder.com/15/f0f8ff/000000?text=+) `Colours.aliceBlue`
- ![#faebd7](https://via.placeholder.com/15/faebd7/000000?text=+) `Colours.antiqueWhite`
- ![#00ffff](https://via.placeholder.com/15/00ffff/000000?text=+) `Colours.aqua`
- ![#7fffd4](https://via.placeholder.com/15/7fffd4/000000?text=+) `Colours.aquamarine`
- ![#f0ffff](https://via.placeholder.com/15/f0ffff/000000?text=+) `Colours.azure`
- ![#f5f5dc](https://via.placeholder.com/15/f5f5dc/000000?text=+) `Colours.beige`
- ![#ffe4c4](https://via.placeholder.com/15/ffe4c4/000000?text=+) `Colours.bisque`
- ![#ffebcd](https://via.placeholder.com/15/ffebcd/000000?text=+) `Colours.blanchedAlmond`
- ![#8a2be2](https://via.placeholder.com/15/8a2be2/000000?text=+) `Colours.blueViolet`
- ![#deb887](https://via.placeholder.com/15/deb887/000000?text=+) `Colours.burlyWood`
- ![#ea7e5d](https://via.placeholder.com/15/ea7e5d/000000?text=+) `Colours.burntSienna`
- ![#5f9ea0](https://via.placeholder.com/15/5f9ea0/000000?text=+) `Colours.cadetBlue`
- ![#7fff00](https://via.placeholder.com/15/7fff00/000000?text=+) `Colours.chartReuse`
- ![#d2691e](https://via.placeholder.com/15/d2691e/000000?text=+) `Colours.chocolate`
- ![#ff7f50](https://via.placeholder.com/15/ff7f50/000000?text=+) `Colours.coral`
- ![#6495ed](https://via.placeholder.com/15/6495ed/000000?text=+) `Colours.cornFlowerBlue`
- ![#fff8dc](https://via.placeholder.com/15/fff8dc/000000?text=+) `Colours.cornSilk`
- ![#dc143c](https://via.placeholder.com/15/dc143c/000000?text=+) `Colours.crimson`
- ![#00008b](https://via.placeholder.com/15/00008b/000000?text=+) `Colours.darkBlue`
- ![#008b8b](https://via.placeholder.com/15/008b8b/000000?text=+) `Colours.darkCyan`
- ![#b8860b](https://via.placeholder.com/15/b8860b/000000?text=+) `Colours.darkGoldenRod`
- ![#006400](https://via.placeholder.com/15/006400/000000?text=+) `Colours.darkGreen`
- ![#a9a9a9](https://via.placeholder.com/15/a9a9a9/000000?text=+) `Colours.darkGrey`
- ![#bdb76b](https://via.placeholder.com/15/bdb76b/000000?text=+) `Colours.darkKhaki`
- ![#8b008b](https://via.placeholder.com/15/8b008b/000000?text=+) `Colours.darkMagenta`
- ![#556b2f](https://via.placeholder.com/15/556b2f/000000?text=+) `Colours.darkOliveGreen`
- ![#ff8c00](https://via.placeholder.com/15/ff8c00/000000?text=+) `Colours.darkOrange`
- ![#9932cc](https://via.placeholder.com/15/9932cc/000000?text=+) `Colours.darkOrchid`
- ![#8b0000](https://via.placeholder.com/15/8b0000/000000?text=+) `Colours.darkRed`
- ![#e9967a](https://via.placeholder.com/15/e9967a/000000?text=+) `Colours.darkSalmon`
- ![#8fbc8f](https://via.placeholder.com/15/8fbc8f/000000?text=+) `Colours.darkSeagreen`
- ![#483d8b](https://via.placeholder.com/15/483d8b/000000?text=+) `Colours.darkSlateBlue`
- ![#2f4f4f](https://via.placeholder.com/15/2f4f4f/000000?text=+) `Colours.darkSlateGrey`
- ![#00ced1](https://via.placeholder.com/15/00ced1/000000?text=+) `Colours.darkTurquoise`
- ![#9400d3](https://via.placeholder.com/15/9400d3/000000?text=+) `Colours.darkViolet`
- ![#ff1493](https://via.placeholder.com/15/ff1493/000000?text=+) `Colours.deepPink`
- ![#00bfff](https://via.placeholder.com/15/00bfff/000000?text=+) `Colours.deepSkyBlue`
- ![#696969](https://via.placeholder.com/15/696969/000000?text=+) `Colours.dimGrey`
- ![#1e90ff](https://via.placeholder.com/15/1e90ff/000000?text=+) `Colours.dodgerBlue`
- ![#b22222](https://via.placeholder.com/15/b22222/000000?text=+) `Colours.fireBrick`
- ![#fffaf0](https://via.placeholder.com/15/fffaf0/000000?text=+) `Colours.floralWhite`
- ![#228b22](https://via.placeholder.com/15/228b22/000000?text=+) `Colours.forestGreen`
- ![ff00ff](https://via.placeholder.com/15/ff00ff/000000?text=+) `Colours.fuchsia`
- ![#dcdcdc](https://via.placeholder.com/15/dcdcdc/000000?text=+) `Colours.gainsboro`
- ![#f8f8ff](https://via.placeholder.com/15/f8f8ff/000000?text=+) `Colours.ghostWhite`
- ![#ffd700](https://via.placeholder.com/15/ffd700/000000?text=+) `Colours.gold`
- ![#daa520](https://via.placeholder.com/15/daa520/000000?text=+) `Colours.goldenRod`
- ![#adff2f](https://via.placeholder.com/15/adff2f/000000?text=+) `Colours.greenYellow`
- ![#f0fff0](https://via.placeholder.com/15/f0fff0/000000?text=+) `Colours.honeyDew`
- ![#ff69b4](https://via.placeholder.com/15/ff69b4/000000?text=+) `Colours.hotPink`
- ![#cd5c5c](https://via.placeholder.com/15/cd5c5c/000000?text=+) `Colours.indianRed`
- ![#fffff0](https://via.placeholder.com/15/fffff0/000000?text=+) `Colours.ivory`
- ![#f0e68c](https://via.placeholder.com/15/f0e68c/000000?text=+) `Colours.khaki`
- ![#e6e6fa](https://via.placeholder.com/15/e6e6fa/000000?text=+) `Colours.lavender`
- ![#fff0f5](https://via.placeholder.com/15/fff0f5/000000?text=+) `Colours.lavenderBlush`
- ![#7cfc00](https://via.placeholder.com/15/7cfc00/000000?text=+) `Colours.lawnGreen`
- ![#fffacd](https://via.placeholder.com/15/fffacd/000000?text=+) `Colours.lemonChiffon`
- ![#f08080](https://via.placeholder.com/15/f08080/000000?text=+) `Colours.lightCoral`
- ![#e0ffff](https://via.placeholder.com/15/e0ffff/000000?text=+) `Colours.lightCyan`
- ![#fafad2](https://via.placeholder.com/15/fafad2/000000?text=+) `Colours.lightGoldenRodYellow`
- ![#d3d3d3](https://via.placeholder.com/15/d3d3d3/000000?text=+) `Colours.lightGrey`
- ![#ffb6c1](https://via.placeholder.com/15/ffb6c1/000000?text=+) `Colours.lightPink`
- ![#ffa07a](https://via.placeholder.com/15/ffa07a/000000?text=+) `Colours.lightSalmon`
- ![#20b2aa](https://via.placeholder.com/15/20b2aa/000000?text=+) `Colours.lightSeaGreen`
- ![#87cefa](https://via.placeholder.com/15/87cefa/000000?text=+) `Colours.lightSkyBlue`
- ![#778899](https://via.placeholder.com/15/778899/000000?text=+) `Colours.lightSlateGrey`
- ![#b0c4de](https://via.placeholder.com/15/b0c4de/000000?text=+) `Colours.lightSteelBlue`
- ![#ffffe0](https://via.placeholder.com/15/ffffe0/000000?text=+) `Colours.lightYellow`
- ![#32cd32](https://via.placeholder.com/15/32cd32/000000?text=+) `Colours.limeGreen`
- ![#faf0e6](https://via.placeholder.com/15/faf0e6/000000?text=+) `Colours.linen`
- ![#ff00ff](https://via.placeholder.com/15/ff00ff/000000?text=+) `Colours.magenta`
- ![#800000](https://via.placeholder.com/15/800000/000000?text=+) `Colours.maroon`
- ![#66cdaa](https://via.placeholder.com/15/66cdaa/000000?text=+) `Colours.mediumAquamarine`
- ![#0000cd](https://via.placeholder.com/15/0000cd/000000?text=+) `Colours.mediumBlue`
- ![#ba55d3](https://via.placeholder.com/15/ba55d3/000000?text=+) `Colours.mediumOrchid`
- ![#9370db](https://via.placeholder.com/15/9370db/000000?text=+) `Colours.mediumPurple`
- ![#3cb371](https://via.placeholder.com/15/3cb371/000000?text=+) `Colours.mediumSeaGreen`
- ![#7b68ee](https://via.placeholder.com/15/7b68ee/000000?text=+) `Colours.mediumSlateBlue`
- ![#00fa9a](https://via.placeholder.com/15/00fa9a/000000?text=+) `Colours.mediumSpringGreen`
- ![#48d1cc](https://via.placeholder.com/15/48d1cc/000000?text=+) `Colours.mediumTurquoise`
- ![#c71585](https://via.placeholder.com/15/c71585/000000?text=+) `Colours.mediumVioletRed`
- ![#191970](https://via.placeholder.com/15/191970/000000?text=+) `Colours.midnightBlue`
- ![#f5fffa](https://via.placeholder.com/15/f5fffa/000000?text=+) `Colours.mintCream`
- ![#ffe4e1](https://via.placeholder.com/15/ffe4e1/000000?text=+) `Colours.mistyRose`
- ![#ffe4b5](https://via.placeholder.com/15/ffe4b5/000000?text=+) `Colours.moccasin`
- ![#ffdead](https://via.placeholder.com/15/ffdead/000000?text=+) `Colours.navajoWhite`
- ![#000080](https://via.placeholder.com/15/000080/000000?text=+) `Colours.navy`
- ![#fdf5e6](https://via.placeholder.com/15/fdf5e6/000000?text=+) `Colours.oldlace`
- ![#808000](https://via.placeholder.com/15/808000/000000?text=+) `Colours.olive`
- ![#6b8e23](https://via.placeholder.com/15/6b8e23/000000?text=+) `Colours.olivedRab`
- ![#ff4500](https://via.placeholder.com/15/ff4500/000000?text=+) `Colours.orangeRed`
- ![#da70d6](https://via.placeholder.com/15/da70d6/000000?text=+) `Colours.orchid`
- ![#eee8aa](https://via.placeholder.com/15/eee8aa/000000?text=+) `Colours.paleGoldenRod`
- ![#98fb98](https://via.placeholder.com/15/98fb98/000000?text=+) `Colours.paleGreen`
- ![#afeeee](https://via.placeholder.com/15/afeeee/000000?text=+) `Colours.paleTurquoise`
- ![#db7093](https://via.placeholder.com/15/db7093/000000?text=+) `Colours.paleVioletRed`
- ![#ffefd5](https://via.placeholder.com/15/ffefd5/000000?text=+) `Colours.papayaWhip`
- ![#ffdab9](https://via.placeholder.com/15/ffdab9/000000?text=+) `Colours.peachPuff`
- ![#cd853f](https://via.placeholder.com/15/cd853f/000000?text=+) `Colours.peru`
- ![#dda0dd](https://via.placeholder.com/15/dda0dd/000000?text=+) `Colours.plum`
- ![#b0e0e6](https://via.placeholder.com/15/b0e0e6/000000?text=+) `Colours.powderBlue`
- ![#663399](https://via.placeholder.com/15/663399/000000?text=+) `Colours.rebeccaPurple`
- ![#bc8f8f](https://via.placeholder.com/15/bc8f8f/000000?text=+) `Colours.rosyBrown`
- ![#4169e1](https://via.placeholder.com/15/4169e1/000000?text=+) `Colours.royalBlue`
- ![#8b4513](https://via.placeholder.com/15/8b4513/000000?text=+) `Colours.saddleBrown`
- ![#fa8072](https://via.placeholder.com/15/fa8072/000000?text=+) `Colours.salmon`
- ![#f4a460](https://via.placeholder.com/15/f4a460/000000?text=+) `Colours.sandyBrown`
- ![#2e8b57](https://via.placeholder.com/15/2e8b57/000000?text=+) `Colours.seaGreen`
- ![#fff5ee](https://via.placeholder.com/15/fff5ee/000000?text=+) `Colours.seaShell`
- ![#a0522d](https://via.placeholder.com/15/a0522d/000000?text=+) `Colours.sienna`
- ![#c0c0c0](https://via.placeholder.com/15/c0c0c0/000000?text=+) `Colours.silver`
- ![#87ceeb](https://via.placeholder.com/15/87ceeb/000000?text=+) `Colours.skyBlue`
- ![#6a5acd](https://via.placeholder.com/15/6a5acd/000000?text=+) `Colours.slateBlue`
- ![#708090](https://via.placeholder.com/15/708090/000000?text=+) `Colours.slateGrey`
- ![#fffafa](https://via.placeholder.com/15/fffafa/000000?text=+) `Colours.snow`
- ![#00ff7f](https://via.placeholder.com/15/00ff7f/000000?text=+) `Colours.springGreen`
- ![#4682b4](https://via.placeholder.com/15/4682b4/000000?text=+) `Colours.steelBlue`
- ![#d2b48c](https://via.placeholder.com/15/d2b48c/000000?text=+) `Colours.tan`
- ![#d8bfd8](https://via.placeholder.com/15/d8bfd8/000000?text=+) `Colours.thistle`
- ![#ff6347](https://via.placeholder.com/15/ff6347/000000?text=+) `Colours.tomato`
- ![#40e0d0](https://via.placeholder.com/15/40e0d0/000000?text=+) `Colours.turquoise`
- ![#ee82ee](https://via.placeholder.com/15/ee82ee/000000?text=+) `Colours.violet`
- ![#f5deb3](https://via.placeholder.com/15/f5deb3/000000?text=+) `Colours.wheat`
- ![#f5f5f5](https://via.placeholder.com/15/f5f5f5/000000?text=+) `Colours.whiteSmoke`
- ![#9acd32](https://via.placeholder.com/15/9acd32/000000?text=+) `Colours.yellowGreen`

## Default Color Swatches

Colours also come with default color swatches from class Colors.

- ![#F44336](https://via.placeholder.com/15/F44336/000000?text=+) `Colours.red`
- ![#FF5252](https://via.placeholder.com/15/FF5252/000000?text=+) `Colours.redAccent`
- ![#E91E63](https://via.placeholder.com/15/E91E63/000000?text=+) `Colours.pink`
- ![#FF4081](https://via.placeholder.com/15/FF4081/000000?text=+) `Colours.pinkAccent`
- ![#9C27B0](https://via.placeholder.com/15/9C27B0/000000?text=+) `Colours.purple`
- ![#E040FB](https://via.placeholder.com/15/E040FB/000000?text=+) `Colours.purpleAccent`
- ![#673AB7](https://via.placeholder.com/15/673AB7/000000?text=+) `Colours.deepPurple`
- ![#7C4DFF](https://via.placeholder.com/15/7C4DFF/000000?text=+) `Colours.deepPurpleAccent`
- ![#3F51B5](https://via.placeholder.com/15/3F51B5/000000?text=+) `Colours.indigo`
- ![#536DFE](https://via.placeholder.com/15/536DFE/000000?text=+) `Colours.indigoAccent`
- ![#2196F3](https://via.placeholder.com/15/2196F3/000000?text=+) `Colours.blue`
- ![#448AFF](https://via.placeholder.com/15/448AFF/000000?text=+) `Colours.blueAccent`
- ![#03A9F4](https://via.placeholder.com/15/03A9F4/000000?text=+) `Colours.lightBlue`
- ![#40C4FF](https://via.placeholder.com/15/40C4FF/000000?text=+) `Colours.lightBlueAccent`
- ![#00BCD4](https://via.placeholder.com/15/00BCD4/000000?text=+) `Colours.cyan`
- ![#18FFFF](https://via.placeholder.com/15/18FFFF/000000?text=+) `Colours.cyanAccent`
- ![#009688](https://via.placeholder.com/15/009688/000000?text=+) `Colours.teal`
- ![#64FFDA](https://via.placeholder.com/15/64FFDA/000000?text=+) `Colours.tealAccent`
- ![#4CAF50](https://via.placeholder.com/15/4CAF50/000000?text=+) `Colours.green`
- ![#69F0AE](https://via.placeholder.com/15/69F0AE/000000?text=+) `Colours.greenAccent`
- ![#8BC34A](https://via.placeholder.com/15/8BC34A/000000?text=+) `Colours.lightGreen`
- ![#B2FF59](https://via.placeholder.com/15/B2FF59/000000?text=+) `Colours.lightGreenAccent`
- ![#CDDC39](https://via.placeholder.com/15/CDDC39/000000?text=+) `Colours.lime`
- ![#EEFF41](https://via.placeholder.com/15/EEFF41/000000?text=+) `Colours.limeAccent`
- ![#FFEB3B](https://via.placeholder.com/15/FFEB3B/000000?text=+) `Colours.yellow`
- ![#FFFF00](https://via.placeholder.com/15/FFFF00/000000?text=+) `Colours.yellowAccent`
- ![#FFC107](https://via.placeholder.com/15/FFC107/000000?text=+) `Colours.amber`
- ![#FFD740](https://via.placeholder.com/15/FFD740/000000?text=+) `Colours.amberAccent`
- ![#FF9800](https://via.placeholder.com/15/FF9800/000000?text=+) `Colours.orange`
- ![#FFAB40](https://via.placeholder.com/15/FFAB40/000000?text=+) `Colours.orangeAccent`
- ![#FF5722](https://via.placeholder.com/15/FF5722/000000?text=+) `Colours.deepOrange`
- ![#FF6E40](https://via.placeholder.com/15/FF6E40/000000?text=+) `Colours.deepOrangeAccent`
- ![#795548](https://via.placeholder.com/15/795548/000000?text=+) `Colours.brown`
- ![#9E9E9E](https://via.placeholder.com/15/9E9E9E/000000?text=+) `Colours.grey`
- ![#607D8B](https://via.placeholder.com/15/607D8B/000000?text=+) `Colours.blueGrey`

Also default colors from Colors:

- `Colours.transparent`

- `Colours.black`
- `Colours.black87`
- `Colours.black54`
- `Colours.black45`
- `Colours.black38`
- `Colours.black26`
- `Colours.black12`

- `Colours.white`
- `Colours.white70`
- `Colours.white60`
- `Colours.white54`
- `Colours.white38`
- `Colours.white30`
- `Colours.white24`
- `Colours.white12`
- `Colours.white10`

## Visual Studio Code Extension 

You can search and install extension named ```fluttercolour``` to show color preview of Colours.

![](assets/fluttercolour1.png)

![](assets/fluttercolour2.png)

CHANGELOG.md
## [0.0.2] - 16 May 2021

* Migrate to null safety

## [0.0.1+1] - 24 Sept 2020

* Update README

## [0.0.1] - 24 Sept 2020

* A working class Colours with more swatches and function swatch to generate customs swatch


lib/src/colours.dart
import 'package:colour/colour.dart';
import 'package:colours/src/generator.dart';
import 'package:flutter/material.dart';

class Colours {
  // This class is not meant to be instantiated or extended; this constructor
  // prevents instantiation and extension.
  Colours._();

  static MaterialColor swatch(dynamic color,
      {ColorsGeneratorAlgorithm algorithm =
          ColorsGeneratorAlgorithm.constantin}) {
    ColorsGenerator generator = ColorsGenerator(algorithm: algorithm);
    return generator
        .generate((color is Color || color is Colour) ? color : Colour(color));
  }

  static MaterialColor aliceBlue = swatch(Colour('f0f8ff'));
  static MaterialColor antiqueWhite = swatch(Colour('faebd7'));
  static MaterialColor aqua = swatch(Colour('00ffff'));
  static MaterialColor aquamarine = swatch(Colour('7fffd4'));
  static MaterialColor azure = swatch(Colour('f0ffff'));
  static MaterialColor beige = swatch(Colour('f5f5dc'));
  static MaterialColor bisque = swatch(Colour('ffe4c4'));
  static MaterialColor blanchedAlmond = swatch(Colour('ffebcd'));
  static MaterialColor blueViolet = swatch(Colour('8a2be2'));
  static MaterialColor burlyWood = swatch(Colour('deb887'));
  static MaterialColor burntSienna = swatch(Colour('ea7e5d'));
  static MaterialColor cadetBlue = swatch(Colour('5f9ea0'));
  static MaterialColor chartReuse = swatch(Colour('7fff00'));
  static MaterialColor chocolate = swatch(Colour('d2691e'));
  static MaterialColor coral = swatch(Colour('ff7f50'));
  static MaterialColor cornFlowerBlue = swatch(Colour('6495ed'));
  static MaterialColor cornSilk = swatch(Colour('fff8dc'));
  static MaterialColor crimson = swatch(Colour('dc143c'));
  static MaterialColor darkBlue = swatch(Colour('00008b'));
  static MaterialColor darkCyan = swatch(Colour('008b8b'));
  static MaterialColor darkGoldenRod = swatch(Colour('b8860b'));
  static MaterialColor darkGreen = swatch(Colour('006400'));
  static MaterialColor darkGrey = swatch(Colour('a9a9a9'));
  static MaterialColor darkKhaki = swatch(Colour('bdb76b'));
  static MaterialColor darkMagenta = swatch(Colour('8b008b'));
  static MaterialColor darkOliveGreen = swatch(Colour('556b2f'));
  static MaterialColor darkOrange = swatch(Colour('ff8c00'));
  static MaterialColor darkOrchid = swatch(Colour('9932cc'));
  static MaterialColor darkRed = swatch(Colour('8b0000'));
  static MaterialColor darkSalmon = swatch(Colour('e9967a'));
  static MaterialColor darkSeagreen = swatch(Colour('8fbc8f'));
  static MaterialColor darkSlateBlue = swatch(Colour('483d8b'));
  static MaterialColor darkSlateGrey = swatch(Colour('2f4f4f'));
  static MaterialColor darkTurquoise = swatch(Colour('00ced1'));
  static MaterialColor darkViolet = swatch(Colour('9400d3'));
  static MaterialColor deepPink = swatch(Colour('ff1493'));
  static MaterialColor deepSkyBlue = swatch(Colour('00bfff'));
  static MaterialColor dimGrey = swatch(Colour('696969'));
  static MaterialColor dodgerBlue = swatch(Colour('1e90ff'));
  static MaterialColor fireBrick = swatch(Colour('b22222'));
  static MaterialColor floralWhite = swatch(Colour('fffaf0'));
  static MaterialColor forestGreen = swatch(Colour('228b22'));
  static MaterialColor fuchsia = swatch(Colour('ff00ff'));
  static MaterialColor gainsboro = swatch(Colour('dcdcdc'));
  static MaterialColor ghostWhite = swatch(Colour('f8f8ff'));
  static MaterialColor gold = swatch(Colour('ffd700'));
  static MaterialColor goldenRod = swatch(Colour('daa520'));
  static MaterialColor greenYellow = swatch(Colour('adff2f'));
  static MaterialColor honeyDew = swatch(Colour('f0fff0'));
  static MaterialColor hotPink = swatch(Colour('ff69b4'));
  static MaterialColor indianRed = swatch(Colour('cd5c5c'));
  static MaterialColor ivory = swatch(Colour('fffff0'));
  static MaterialColor khaki = swatch(Colour('f0e68c'));
  static MaterialColor lavender = swatch(Colour('e6e6fa'));
  static MaterialColor lavenderBlush = swatch(Colour('fff0f5'));
  static MaterialColor lawnGreen = swatch(Colour('7cfc00'));
  static MaterialColor lemonChiffon = swatch(Colour('fffacd'));
  static MaterialColor lightCoral = swatch(Colour('f08080'));
  static MaterialColor lightCyan = swatch(Colour('e0ffff'));
  static MaterialColor lightGoldenRodYellow = swatch(Colour('fafad2'));
  static MaterialColor lightGrey = swatch(Colour('d3d3d3'));
  static MaterialColor lightPink = swatch(Colour('ffb6c1'));
  static MaterialColor lightSalmon = swatch(Colour('ffa07a'));
  static MaterialColor lightSeaGreen = swatch(Colour('20b2aa'));
  static MaterialColor lightSkyBlue = swatch(Colour('87cefa'));
  static MaterialColor lightSlateGrey = swatch(Colour('778899'));
  static MaterialColor lightSteelBlue = swatch(Colour('b0c4de'));
  static MaterialColor lightYellow = swatch(Colour('ffffe0'));
  static MaterialColor limeGreen = swatch(Colour('32cd32'));
  static MaterialColor linen = swatch(Colour('faf0e6'));
  static MaterialColor magenta = swatch(Colour('ff00ff'));
  static MaterialColor maroon = swatch(Colour('800000'));
  static MaterialColor mediumAquamarine = swatch(Colour('66cdaa'));
  static MaterialColor mediumBlue = swatch(Colour('0000cd'));
  static MaterialColor mediumOrchid = swatch(Colour('ba55d3'));
  static MaterialColor mediumPurple = swatch(Colour('9370db'));
  static MaterialColor mediumSeaGreen = swatch(Colour('3cb371'));
  static MaterialColor mediumSlateBlue = swatch(Colour('7b68ee'));
  static MaterialColor mediumSpringGreen = swatch(Colour('00fa9a'));
  static MaterialColor mediumTurquoise = swatch(Colour('48d1cc'));
  static MaterialColor mediumVioletRed = swatch(Colour('c71585'));
  static MaterialColor midnightBlue = swatch(Colour('191970'));
  static MaterialColor mintCream = swatch(Colour('f5fffa'));
  static MaterialColor mistyRose = swatch(Colour('ffe4e1'));
  static MaterialColor moccasin = swatch(Colour('ffe4b5'));
  static MaterialColor navajoWhite = swatch(Colour('ffdead'));
  static MaterialColor navy = swatch(Colour('000080'));
  static MaterialColor oldlace = swatch(Colour('fdf5e6'));
  static MaterialColor olive = swatch(Colour('808000'));
  static MaterialColor olivedRab = swatch(Colour('6b8e23'));
  static MaterialColor orangeRed = swatch(Colour('ff4500'));
  static MaterialColor orchid = swatch(Colour('da70d6'));
  static MaterialColor paleGoldenRod = swatch(Colour('eee8aa'));
  static MaterialColor paleGreen = swatch(Colour('98fb98'));
  static MaterialColor paleTurquoise = swatch(Colour('afeeee'));
  static MaterialColor paleVioletRed = swatch(Colour('db7093'));
  static MaterialColor papayaWhip = swatch(Colour('ffefd5'));
  static MaterialColor peachPuff = swatch(Colour('ffdab9'));
  static MaterialColor peru = swatch(Colour('cd853f'));
  static MaterialColor plum = swatch(Colour('dda0dd'));
  static MaterialColor powderBlue = swatch(Colour('b0e0e6'));
  static MaterialColor rebeccaPurple = swatch(Colour('663399'));
  static MaterialColor rosyBrown = swatch(Colour('bc8f8f'));
  static MaterialColor royalBlue = swatch(Colour('4169e1'));
  static MaterialColor saddleBrown = swatch(Colour('8b4513'));
  static MaterialColor salmon = swatch(Colour('fa8072'));
  static MaterialColor sandyBrown = swatch(Colour('f4a460'));
  static MaterialColor seaGreen = swatch(Colour('2e8b57'));
  static MaterialColor seaShell = swatch(Colour('fff5ee'));
  static MaterialColor sienna = swatch(Colour('a0522d'));
  static MaterialColor silver = swatch(Colour('c0c0c0'));
  static MaterialColor skyBlue = swatch(Colour('87ceeb'));
  static MaterialColor slateBlue = swatch(Colour('6a5acd'));
  static MaterialColor slateGrey = swatch(Colour('708090'));
  static MaterialColor snow = swatch(Colour('fffafa'));
  static MaterialColor springGreen = swatch(Colour('00ff7f'));
  static MaterialColor steelBlue = swatch(Colour('4682b4'));
  static MaterialColor tan = swatch(Colour('d2b48c'));
  static MaterialColor thistle = swatch(Colour('d8bfd8'));
  static MaterialColor tomato = swatch(Colour('ff6347'));
  static MaterialColor turquoise = swatch(Colour('40e0d0'));
  static MaterialColor violet = swatch(Colour('ee82ee'));
  static MaterialColor wheat = swatch(Colour('f5deb3'));
  static MaterialColor whiteSmoke = swatch(Colour('f5f5f5'));
  static MaterialColor yellowGreen = swatch(Colour('9acd32'));

  static const MaterialColor red = Colors.red;
  static const MaterialAccentColor redAccent = Colors.redAccent;
  static const MaterialColor pink = Colors.pink;
  static const MaterialAccentColor pinkAccent = Colors.pinkAccent;
  static const MaterialColor purple = Colors.purple;
  static const MaterialAccentColor purpleAccent = Colors.purpleAccent;
  static const MaterialColor deepPurple = Colors.deepPurple;
  static const MaterialAccentColor deepPurpleAccent = Colors.deepPurpleAccent;
  static const MaterialColor indigo = Colors.indigo;
  static const MaterialAccentColor indigoAccent = Colors.indigoAccent;
  static const MaterialColor blue = Colors.blue;
  static const MaterialAccentColor blueAccent = Colors.blueAccent;
  static const MaterialColor lightBlue = Colors.lightBlue;
  static const MaterialAccentColor lightBlueAccent = Colors.lightBlueAccent;
  static const MaterialColor cyan = Colors.cyan;
  static const MaterialAccentColor cyanAccent = Colors.cyanAccent;
  static const MaterialColor teal = Colors.teal;
  static const MaterialAccentColor tealAccent = Colors.tealAccent;
  static const MaterialColor green = Colors.green;
  static const MaterialAccentColor greenAccent = Colors.greenAccent;
  static const MaterialColor lightGreen = Colors.green;
  static const MaterialAccentColor lightGreenAccent = Colors.lightGreenAccent;
  static const MaterialColor lime = Colors.lime;
  static const MaterialAccentColor limeAccent = Colors.limeAccent;
  static const MaterialColor yellow = Colors.yellow;
  static const MaterialAccentColor yellowAccent = Colors.yellowAccent;
  static const MaterialColor amber = Colors.amber;
  static const MaterialAccentColor amberAccent = Colors.amberAccent;
  static const MaterialColor orange = Colors.orange;
  static const MaterialAccentColor orangeAccent = Colors.orangeAccent;
  static const MaterialColor deepOrange = Colors.deepOrange;
  static const MaterialAccentColor deepOrangeAccent = Colors.deepOrangeAccent;
  static const MaterialColor brown = Colors.brown;
  static const MaterialColor grey = Colors.grey;
  static const MaterialColor blueGrey = Colors.blueGrey;

  /// The material design primary color swatches, excluding grey.
  static const List<MaterialColor> primaries = Colors.primaries;

  /// The material design accent color swatches.
  static const List<MaterialAccentColor> accents = Colors.accents;

  static const Color transparent = Colors.transparent;
  static const Color black = Colors.black;
  static const Color black87 = Colors.black87;
  static const Color black54 = Colors.black54;
  static const Color black45 = Colors.black45;
  static const Color black38 = Colors.black38;
  static const Color black26 = Colors.black26;
  static const Color black12 = Colors.black12;

  static const Color white = Colors.white;
  static const Color white70 = Colors.white70;
  static const Color white60 = Colors.white60;
  static const Color white54 = Colors.white54;
  static const Color white38 = Colors.white38;
  static const Color white30 = Colors.white30;
  static const Color white24 = Colors.white24;
  static const Color white12 = Colors.white12;
  static const Color white10 = Colors.white10;
}
